Strata/DDM/Integration/Java/Gen.lean

```diff
@@ -0,0 +1,500 @@
+
+inductive JavaType where
+  | simple (name : String) (boxed : Option String := none)
+  | array (elem : JavaType)
+  | optional (elem : JavaType)
+  | list (elem : JavaType)
+  deriving Inhabited
+
+mutual
+def JavaType.toJava : JavaType → String
+  | .simple name _ => name
+  | .array elem => elem.toJava ++ "[]"
+  | .optional elem => s!"java.util.Optional<{elem.toJavaBoxed}>"
+  | .list elem => s!"java.util.List<{elem.toJavaBoxed}>"
+
+def JavaType.toJavaBoxed : JavaType → String
+  | .simple _ (some boxed) => boxed
+  | t => t.toJava
+end
+
+partial def syntaxCatToJavaType (cat : SyntaxCat) : JavaType :=
+  match cat.name with
+  | ⟨"Init", "Ident"⟩ => .simple "java.lang.String"
+  | ⟨"Init", "Num"⟩ => .simple "java.math.BigInteger"
+  | ⟨"Init", "Decimal"⟩ => .simple "java.math.BigDecimal"
+  | ⟨"Init", "Str"⟩ => .simple "java.lang.String"
+  | ⟨"Init", "ByteArray"⟩ => .array (.simple "byte" (some "java.lang.Byte"))
+  | ⟨"Init", "Bool"⟩ => .simple "boolean" (some "java.lang.Boolean")
+  | ⟨"Init", "Option"⟩ =>
+    match cat.args[0]? with
+    | some inner => .optional (syntaxCatToJavaType inner)
+    | none => .optional (.simple "java.lang.Object")
+  | ⟨"Init", "Seq"⟩ | ⟨"Init", "CommaSepBy"⟩ =>
+    match cat.args[0]? with
+    | some inner => .list (syntaxCatToJavaType inner)
+    | none => .list (.simple "java.lang.Object")
+  | ⟨"Init", "Expr"⟩ => .simple "Expr"
+  | ⟨"Init", "TypeExpr"⟩ => .simple "TypeExpr"
+  | ⟨"Init", "Type"⟩ => .simple "Type_"
+  | ⟨"Init", "TypeP"⟩ => .simple "TypeP"
+  | ⟨_, name⟩ => .simple (escapeJavaName (toPascalCase name))
+
+def argDeclKindToJavaType : ArgDeclKind → JavaType
+  | .type _ => .simple "Expr"
+  | .cat c => syntaxCatToJavaType c
+
+/-- Extract the Java interface name from a SyntaxCat, or none if it maps to a primitive -/
+partial def syntaxCatToInterfaceName (cat : SyntaxCat) : Option String :=
+  match cat.name with
+  -- Primitives map to Java types, no interface needed
+  | ⟨"Init", "Ident"⟩ | ⟨"Init", "Num"⟩ | ⟨"Init", "Decimal"⟩
+  | ⟨"Init", "Str"⟩ | ⟨"Init", "ByteArray"⟩ | ⟨"Init", "Bool"⟩ => none
+  -- Containers - recurse into element type
+  | ⟨"Init", "Option"⟩ | ⟨"Init", "Seq"⟩ | ⟨"Init", "CommaSepBy"⟩ =>
+    cat.args[0]?.bind syntaxCatToInterfaceName
+  -- Abstract Init categories (extension points for dialects)
+  | ⟨"Init", "Expr"⟩ => some "Expr"
+  | ⟨"Init", "TypeExpr"⟩ => some "TypeExpr"
+  | ⟨"Init", "Type"⟩ => some "Type_"
+  | ⟨"Init", "TypeP"⟩ => some "TypeP"
+  -- Other Init types are internal DDM machinery
+  | ⟨"Init", _⟩ => none
+  -- Dialect-defined categories
+  | ⟨_, name⟩ => some (escapeJavaName (toPascalCase name))
+
+/-! ## Java Structures -/
+
+structure JavaField where
+  name : String
+  type : JavaType
+
+structure JavaRecord where
+  name : String
+  operationName : QualifiedIdent
+  implements : String
+  fields : Array JavaField
+
+structure JavaInterface where
+  name : String
+  permits : Array String
+
+structure GeneratedFiles where
```

Comment:
I think some comments on these datatypes would be useful; particularly `GeneratedFiles` and `NameAssignments`.